src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/MeteredEmbeddingProvider.java

```diff
@@ -54,6 +57,21 @@ public Uni<Response> vectorize(
       List<String> texts,
       EmbeddingCredentials embeddingCredentials,
       EmbeddingRequestType embeddingRequestType) {
+
+    // Validate if the model is END_OF_LIFE
+    if (embeddingProvider.model.modelSupport().status() == ModelSupport.SupportStatus.END_OF_LIFE) {
+      throw SchemaException.Code.UNSUPPORTED_PROVIDER_MODEL.get(
+          Map.of(
+              "model", embeddingProvider.model.name(),
+              "modelStatus", embeddingProvider.model.modelSupport().status().name(),
+              "message",
+                  embeddingProvider
+                      .model
+                      .modelSupport()
+                      .message()
+                      .orElse("The model is not supported.")));
```

Comment:
Maybe change default slightly to something like:

"The model is no longer supported (status: END_OF_LIFE)"